kotlinx-coroutines-core/common/src/sync/Mutex.kt

```diff
@@ -361,24 +358,28 @@ internal class MutexImpl(locked: Boolean) : Mutex, SelectClause2<Any?, Mutex> {
     }
 
     private class LockCont(
+        mutex: MutexImpl,
```

Comment:
can be plain `Mutex` here

Reply:
LockCont is an inner class now, not an issue anymore.